scripts/ninja.js

```diff
@@ -585,25 +586,53 @@ function sourceToTarget(y) {
  */
 function ocamlDepForBscAsync(files, dir, depsMap) {
   return new Promise((resolve, reject) => {
+    var tmpdir = null;
+    const mlfiles = []; // convert .res files to temporary .ml files in tmpdir
+    files.forEach(f => {
+      const { name, ext } = path.parse(f);
+      if (ext === ".res" || ext === ".resi") {
+        const mlname = ext === ".resi" ? name + ".mli" : name + ".ml";
+        if (tmpdir == null) {
+          tmpdir = fs.mkdtempSync(path.join(os.tmpdir(), "resToMl"));
+        }
+        try {
+          const mlfile = path.join(tmpdir, mlname);
+          cp.execSync(`${bsc_exe} -dsource -only-parse ${f} 2>${mlfile}`, {
+            cwd: dir,
+            shell: true,
```

Comment:
nit: the shell option property seems string instead bool. https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#child_processexecsynccommand-options

Reply:
> shell [<string>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#String_type) Shell to execute the command with. See [Shell requirements](https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#shell-requirements) and [Default Windows shell](https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#default-windows-shell). Default: '/bin/sh' on Unix, process.env.ComSpec on Windows.

I guess we don't need shell property here. Can we omit it?

Reply:
Actually making `shell: "true"` broke it.
Fixed it back.

Reply:
I guess I misunderstood the comment initially.